### PR TITLE
Adding swagger changes for Virtual Network Rules

### PR DESCRIPTION
**Pull request by dnayantara**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Could you update description to something more specific to VNetRules?

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: VnetFirewallRules

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Do you return 202?  If so, I think you'll need to add 
"202": {
"description": "Accepted"
}

to responses.  No need to include the response body for 202 that in the swagger or examples though, just the response code with an empty body.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Are the Operations definitions being used?

**Review by nathannfan — CHANGES_REQUESTED** [review]
Looks good!  A few minor changes for it to work for SDK, but it's close.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: I'm assuming this operation follows azure async pattern, which returns 202 initially?

Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Yes it does. I will add that.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: I dont think so. I didnt think we would have to do anything separately for the operations

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Sure. Makes sense

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: You're right, we only need them in one file per API version.  You can remove all the Operations definitions then.

Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Got it.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Sorry, I should clarify.  VnetFirewallRules_Get - we've been going plural_operation for operation id

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: This should also clear up one of the validation errors

**Comment by nathannfan** [comment]
@dnayantara Looks like the validation failures are because of parameter description mismatches.  Copying the parameters from https://github.com/Azure/azure-rest-api-specs/blob/master/arm-sql/2015-05-01-preview/swagger/blobAuditingPolicies.json should fix it.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: nit: Maintain parameter ordering, place ```SubscriptionId``` and ```ApiVersion``` together at the beginning or the end of the parameters section

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: nit: parameter ordering

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: nit: parameter ordering

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: nit: parameter ordering

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Looks like a Resource, there are a couple of options here either reference [this](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-sql/2015-05-01-preview/swagger/blobAuditingPolicies.json#L123) model or redefine the same structure here and do an allOf. (I'd recommend not redefining simply since its duplication and overhead)

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Should this be readonly? Or do you allow this as a part of the request object?

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: set this to readonly

**Review by dsgouda — CHANGES_REQUESTED** [review]
Requesting minor changes

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: should be `VnetFirewallRules_ListByServer`

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Yes, please copy Resource+TrackedResource/ProxyResource from other SQL files and use allOf to include TrackedResource/ProxyResource as appropriate.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Add x-ms-flatten:true

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: `VnetFirewallRuleListResult`

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Do you actually have paging? If no, remove this and add x-ms-paging: { nextLinkName: null } to the list operation.

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: This is allowed as part of the request object. I dont think this is readonly

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Is this a proxy resource? If so, remove location, id, name, and type from this definition and use allOf, e.g. https://github.com/Azure/azure-rest-api-specs/blob/master/arm-sql/2014-04-01/swagger/sql.core.json#L1627

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Should be named `VnetFirewallRuleProperties`

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Formatting looks a little off

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: What @jaredmoo said

**Comment by jaredmoo** [comment]
FYI the ListByOperationsValidation linter error regarding ListByServer should be ignored. See discussion with @veronicagg at https://github.com/Azure/azure-rest-api-specs/pull/1005#issuecomment-285130957

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: nit: What I implied was cross-referencing using definitions that already exist under your RP like so
https://github.com/Azure/azure-rest-api-specs/blob/master/arm-recoveryservices/2016-06-01/swagger/replicationusages.json#L46
This is not a requirement but reduces duplication

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Not making this change as @jaredmoo said in the email, cross references don't work.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: This should be ```x-ms-client-flatten```

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: This should be ```x-ms-pageable```, please recheck extension names

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Replied on the email thread. However, won't block on this.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Ill change it to virtual network rules since thats what we are calling it to the customers.

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by dnayantara** [comment]
I will add another iteration which has been directly generated from the source code soon.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Please mark this as readonly

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: ```name:parameters``` makes sense for a body parameter. This is correct

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: nit: Just want to point out that the VnetFirewallRule model has no required properties which means it could theoretically be an empty object, please verify that the service is fine with this.

**Review by dsgouda — APPROVED** [review]
Couple of minor comments and a suggestion. LGTM pending CI builds

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: No. the parameters should be required. I will change this.

File: arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Did you plan to add required properties for VnetFirewallRule in this commit?

Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: Lets just leave it like this for now. I may make changes later.

Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: I'm fine with it as long as the service is OK with an empty body parameter

Comment (dnayantara) [inline arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json]: I am with the service team and we are ok with this for now. Please accept pull request if everything else looks good.

**Pull request merged**
